plugins/languages/typescript-tools.nix

```diff
@@ -0,0 +1,177 @@
+{
+  lib,
+  pkgs,
+  helpers,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.typescript-tools;
+in {
+  options.plugins.typescript-tools = {
+    enable = mkEnableOption "typescript-tools";
+    package =
+      helpers.mkPackageOption "typescript-tools"
+      pkgs.vimPlugins.typescript-tools-nvim;
+
+    handlers = mkOption {
+      type = with helpers.nixvimTypes; nullOr (attrsOf strLuaFn);
+      apply = v: helpers.ifNonNull' v (mapAttrs (_: helpers.mkRaw) v);
+      default = null;
+      description = "How tsserver should respond to LSP requests";
+      example = {
+        "textDocument/publishDiagnostics" = ''
+          require("typescript-tools.api").filter_diagnostics(
+            -- Ignore 'This may be converted to an async function' diagnostics.
+            { 80006 }
+          )
+        '';
+      };
+    };
+
+    settings = {
+      separateDiagnosticServer =
+        helpers.defaultNullOpts.mkBool true
+        "Spawns an additional tsserver instance to calculate diagnostics";
+
+      publishDiagnosticOn = mkOption {
+        type = with types; enum ["change" "insert_leave"];
+        default = "insert_leave";
+        description = ''
+          Either "change" or "insert_leave". Determines when the client asks the server about diagnostics'';
+      };
+
+      exposeAsCodeAction = mkOption {
+        type = with types;
+          either (enum ["all"]) (enum [
+            "fix_all"
+            "add_missing_imports"
+            "remove_unused"
+            "remove_unused_imports"
+            "organize_imports"
+            "insert_leave"
+          ]);
+        default = [];
+        description = "Specify what to expose as code actions.";
+      };
+
+      tsserverPath = helpers.mkNullOrOption types.str ''
```

Comment:
helpers.mkNullOrStr, this allows the user to provide raw lua values instead of a string if he requires it